src/Core/Resolvers/SqlQueryEngine.cs

```diff
@@ -186,20 +187,45 @@ private OkObjectResult FormatFindResult(JsonDocument jsonDoc, FindRequestContext
         {
             JsonElement jsonElement = jsonDoc.RootElement.Clone();
 
+            // When there are no rows returned from the database, the jsonElement will be an empty array.
+            // In that case, the response is returned as is.
+            if (jsonElement.ValueKind is JsonValueKind.Array && jsonElement.GetArrayLength() == 0)
+            {
+                return OkResponse(jsonElement);
+            }
+
+            IEnumerable<string> extraFieldsInResponse = (jsonElement.ValueKind is not JsonValueKind.Array)
+                                                  ? DetermineExtraFieldsInResponse(jsonElement, context)
+                                                  : DetermineExtraFieldsInResponse(jsonElement.EnumerateArray().First(), context);
+
+            int countOfExtraFieldsInResponse = extraFieldsInResponse.Count();
+
             // If the results are not a collection or if the query does not have a next page
-            // no nextLink is needed, return JsonDocument as is
+            // no nextLink is needed. So, the response is returned after removing the extra fields.
             if (jsonElement.ValueKind is not JsonValueKind.Array || !SqlPaginationUtil.HasNext(jsonElement, context.First))
             {
-                // Clones the root element to a new JsonElement that can be
-                // safely stored beyond the lifetime of the original JsonDocument.
-                return OkResponse(jsonElement);
+                // If there are no additional fields present, the response is returned directly. When there
+                // are extra fields, they are removed before returning the response.
+                if (countOfExtraFieldsInResponse == 0)
+                {
+                    return OkResponse(jsonElement);
+                }
+                else
+                {
+                    return jsonElement.ValueKind is JsonValueKind.Array ? OkResponse(JsonSerializer.SerializeToElement(RemoveExtraFieldsInResponseWithMultipleItems(jsonElement.EnumerateArray().ToList(), extraFieldsInResponse)))
+                                                                        : OkResponse(RemoveExtraFieldsInResponseWithSingleItem(jsonElement, extraFieldsInResponse));
+                }
             }
 
-            // More records exist than requested, we know this by requesting 1 extra record,
-            // that extra record is removed here.
             IEnumerable<JsonElement> rootEnumerated = jsonElement.EnumerateArray();
```

Comment:
Would be better to just have this as a list since .Count() and .Last() and both traverse the enumerable.

Reply:
I've updated to use List and am performing index based access which is O(1). [List docs](https://learn.microsoft.com/dotnet/api/system.collections.generic.list-1.item?#remarks)